File: CS/WindowsApplication35/Form1.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.XtraReports.UI;
// ...

namespace WindowsApplication35 {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e) {
            XtraReport1 report = new XtraReport1();
            report.DataSource = CreateData();
            ReportPrintTool printTool = new ReportPrintTool(report);
            printTool.ShowPreviewDialog();
        }

        private List<Data> CreateData() {
            List<Data> data = new List<Data>();

            Data item1 = new Data();
            item1.Date = DateTime.Now;
            item1.Id = 0;
            item1.Name = "First";
            data.Add(item1);

            Data item2 = new Data();
            item2.Date = DateTime.Now;
            item2.Id = 1;
            item2.Name = "Second";
            data.Add(item2);

            Data item3 = new Data();
            item3.Date = DateTime.Now;
            item3.Id = 2;
            item3.Name = "Third";
            data.Add(item3);

            return data;
        }

    }
}

File: VB/WindowsApplication35/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports System.Collections.Generic
' ...

Namespace WindowsApplication35
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			Dim report As New XtraReport1()
			report.DataSource = CreateData()
			report.ShowPreview()
		End Sub

		Private Function CreateData() As List(Of Data)
			Dim data As New List(Of Data)()

			Dim item1 As New Data()
			item1.Date = DateTime.Now
			item1.Id = 0
			item1.Name = "First"
			data.Add(item1)

			Dim item2 As New Data()
			item2.Date = DateTime.Now
			item2.Id = 1
			item2.Name = "Second"
			data.Add(item2)

			Dim item3 As New Data()
			item3.Date = DateTime.Now
			item3.Id = 2
			item3.Name = "Third"
			data.Add(item3)

			Return data
		End Function

	End Class
End Namespace